File: app/src/main/java/com/wapchief/imitationlol/NestedScrollableViewHelper.java
package com.wapchief.imitationlol;

import android.support.v4.widget.NestedScrollView;
import android.view.View;
import android.widget.ScrollView;

import com.sothree.slidinguppanel.ScrollableViewHelper;

/**
 * Created by apple on 2017/6/28.
 */

public class NestedScrollableViewHelper extends ScrollableViewHelper {
    public int getScrollableViewScrollPosition(View scrollableView, boolean isSlidingUp) {
        View mScrollableView = null;
        if (mScrollableView instanceof NestedScrollView) {
            if(isSlidingUp){
                return mScrollableView.getScrollY();
            } else {
                NestedScrollView nsv = ((NestedScrollView) mScrollableView);
                View child = nsv.getChildAt(0);
                return (child.getBottom() - (nsv.getHeight() + nsv.getScrollY()));
            }
        } else {
            return 0;
        }
    }
}

File: app/src/main/java/com/wapchief/imitationlol/NestedScrollableViewHelper.kt
package com.wapchief.imitationlol

import android.view.View
import androidx.core.widget.NestedScrollView
import com.sothree.slidinguppanel.ScrollableViewHelper

/**
 * Created by apple on 2017/6/28.
 */
class NestedScrollableViewHelper : ScrollableViewHelper() {
    override fun getScrollableViewScrollPosition(scrollableView: View, isSlidingUp: Boolean): Int {
        val mScrollableView: View? = null
        return if (mScrollableView is NestedScrollView) {
            if (isSlidingUp) {
                mScrollableView.getScrollY()
            } else {
                val nsv = mScrollableView
                val child = nsv.getChildAt(0)
                child.bottom - (nsv.height + nsv.scrollY)
            }
        } else {
            0
        }
    }
}